packages/roosterjs-content-model-plugins/lib/markdown/utils/setFormat.ts

```diff
@@ -0,0 +1,90 @@
+        );
+
+        if (selectedSegmentsAndParagraphs.length > 0 && selectedSegmentsAndParagraphs[0][1]) {
+            const marker = selectedSegmentsAndParagraphs[0][0];
+            context.newPendingFormat = {
+                ...marker.format,
+                strikethrough: !!marker.format.strikethrough,
+                italic: !!marker.format.italic,
+                fontWeight: marker.format?.fontWeight ? 'bold' : undefined,
+            };
+
+            const paragraph = selectedSegmentsAndParagraphs[0][1];
+            if (marker.segmentType == 'SelectionMarker') {
+                const markerIndex = paragraph.segments.indexOf(marker);
+                let previousCharacterSegment: ContentModelText | undefined = undefined;
+                let index = markerIndex;
+                if (markerIndex > 0 && paragraph.segments[markerIndex - 1]) {
+                    const segmentBeforeMarker = paragraph.segments[markerIndex - 1];
+                    if (
+                        segmentBeforeMarker.segmentType == 'Text' &&
+                        segmentBeforeMarker.text[segmentBeforeMarker.text.length - 1] == character
+                    ) {
+                        for (let i = markerIndex - 1; i >= 0; i--) {
+                            const previousSegment = paragraph.segments[i];
+                            if (
+                                previousSegment &&
+                                previousSegment.segmentType == 'Text' &&
+                                previousSegment.text.indexOf(character) > -1 &&
+                                !(
+                                    i == markerIndex - 1 &&
+                                    previousSegment.text.indexOf(character) ==
+                                        previousSegment.text.length - 1
+                                )
+                            ) {
+                                previousCharacterSegment = previousSegment;
+                                index = i;
+                                break;
+                            }
+                        }
+                        if (previousCharacterSegment && index < markerIndex) {
+                            const splittedSegments = previousCharacterSegment.text.split(character);
+
+                            const formattedText = splittedSegments[splittedSegments.length - 2];
+                            if (!formattedText) {
+                                return false;
+                            }
+
+                            previousCharacterSegment.text = previousCharacterSegment.text.substring(
+                                0,
+                                previousCharacterSegment.text.length - formattedText.length - 2
+                            );
+
+                            const formattedSegment = createText(formattedText, marker.format);
+                            paragraph.segments.splice(index + 1, 0, formattedSegment);
```

Comment:
If you can create a common function to help split the text segment (as I suggested in another PR), you can leverage here to make this easier

Reply:
I marked this PR as draft, since I'll will this one to merged with the `splitTextSegment` https://github.com/microsoft/roosterjs/pull/2522